packages/config/src/__tests__/loadConfigFromFile.test.ts

```diff
@@ -204,45 +274,314 @@ describe('loadConfigFromFile', () => {
+      expect(config).toMatchObject({
+        earlyAccess: true,
+        loadedFromFile: resolvedPath,
+      })
+    })
+
+    it('prisma.config.mjs is 3rd choice', async () => {
+      ctx.fixture('loadConfigFromFile/precedence')
+      await Promise.all([ctx.fs.removeAsync('prisma.config.js'), ctx.fs.removeAsync('prisma.config.ts')])
+
+      const { config, error, resolvedPath } = await loadConfigFromFile({})
+      expect(resolvedPath).toMatch(path.join(ctx.fs.cwd(), 'prisma.config.mjs'))
+      expect(error).toBeUndefined()
+      expect(config).toMatchObject({
+        earlyAccess: true,
+        loadedFromFile: resolvedPath,
+      })
+    })
+
+    it('prisma.config.cjs is 4th choice', async () => {
+      ctx.fixture('loadConfigFromFile/precedence')
+      await Promise.all([
+        ctx.fs.removeAsync('prisma.config.js'),
+        ctx.fs.removeAsync('prisma.config.ts'),
+        ctx.fs.removeAsync('prisma.config.mjs'),
+      ])
+
+      const { config, error, resolvedPath } = await loadConfigFromFile({})
+      expect(resolvedPath).toMatch(path.join(ctx.fs.cwd(), 'prisma.config.cjs'))
+      expect(error).toBeUndefined()
+      expect(config).toMatchObject({
+        earlyAccess: true,
+        loadedFromFile: resolvedPath,
+      })
+    })
+
+    it('prisma.config.mts is 5th choice', async () => {
+      ctx.fixture('loadConfigFromFile/precedence')
+      await Promise.all([
+        ctx.fs.removeAsync('prisma.config.js'),
+        ctx.fs.removeAsync('prisma.config.ts'),
+        ctx.fs.removeAsync('prisma.config.mjs'),
+        ctx.fs.removeAsync('prisma.config.cjs'),
+      ])
+
+      const { config, error, resolvedPath } = await loadConfigFromFile({})
+      expect(resolvedPath).toMatch(path.join(ctx.fs.cwd(), 'prisma.config.mts'))
+      expect(error).toBeUndefined()
+      expect(config).toMatchObject({
+        earlyAccess: true,
+        loadedFromFile: resolvedPath,
+      })
+    })
+
+    it('prisma.config.cts is 6th choice', async () => {
+      ctx.fixture('loadConfigFromFile/precedence')
+      await Promise.all([
+        ctx.fs.removeAsync('prisma.config.js'),
+        ctx.fs.removeAsync('prisma.config.ts'),
+        ctx.fs.removeAsync('prisma.config.mjs'),
+        ctx.fs.removeAsync('prisma.config.cjs'),
+        ctx.fs.removeAsync('prisma.config.mts'),
+      ])
+
+      const { config, error, resolvedPath } = await loadConfigFromFile({})
+      expect(resolvedPath).toMatch(path.join(ctx.fs.cwd(), 'prisma.config.cts'))
+      expect(error).toBeUndefined()
+      expect(config).toMatchObject({
+        earlyAccess: true,
+        loadedFromFile: resolvedPath,
+      })
+    })
+
+    // Note: As of c12@3.1.0, it tries loading `.json` even when such extension is excluded
+    // from `jiti` or `jitiOptions.extensions`.
+    // See: https://github.com/unjs/c12/blob/1efbcbce0e094a8f8a0ba676324affbef4a0ba8b/src/loader.ts#L443.
+    it('prisma.config.json is 7th choice', async () => {
```

Comment:
Fyi, `prisma.config.json` is currently potentially loaded even though we don't support it.
Handling this can happen in a follow-up PR that will use a modified version of `c12`.